packages/astro/src/core/config/settings.ts

```diff
@@ -20,5 +20,6 @@ export function createSettings({ config, tsConfig, tsConfigPath }: CreateSetting
 		pageExtensions: ['.astro', '.md', '.html'],
 		renderers: [jsxRenderer],
 		scripts: [],
+		watchTargets: tsConfigPath ? [{ path: tsConfigPath, type: 'absolute' }] : [],
```

Comment:
for `tsconfig.json`, you probably want more than just the main path, as you also want to restart if a tsconfig.json that is extended from gets changed. `tsconfig-resolver` returns this in `extendedPaths`

Reply:
I've updated it, what do you think?